code/game/objects/structures/janicart.dm

```diff
@@ -32,7 +34,6 @@
 	myreplacer = new /obj/item/device/lightreplacer(src)
 
 	mybucket = new /obj/structure/mopbucket(src)
-	mybucket.reagents.add_reagent(/decl/reagent/water, mybucket.bucketsize)
```

Comment:
Have you intentionally removed filling the bucket with water when a cart is spawned?
(Imho thats a decent QOL feature and it makes it easier to spawn in "full" carts)

Reply:
it's intentional, so that you can fill the bucket with whatever.